localstack/services/generic_proxy.py

```diff
@@ -592,136 +594,88 @@ def get_flask_ssl_context(cls, serial_number=None):
         return None
 
 
-class UrlMatchingForwarder(ProxyListener):
+class RegisteredProxyServer(Server, abc.ABC):
```

Comment:
I think this should be moved to the opensearch package (since it's not used by anyone else yet).
And the name might be a bit confusing, maybe just `ProxyServer`?

Reply:
I was unsure about that, since the previous class also was rather generic, but still only used by Opensearch. Agreed on the naming, it is weird. Not sure if a generic `ProxyServer` would fit in the opensearch package namewise, you would expect something generic like this in a package like http. Naming is hard